tests/CppWinRTAuthoringTests.cpp

```diff
@@ -206,57 +206,70 @@ TEST_CASE("CppWinRTAuthoringTests::EventsAndCppWinRt", "[property]")
-    }
-    {
-        struct Test : winrt::implements<Test, winrt::Windows::UI::Xaml::Data::INotifyPropertyChanged>, wil::notify_property_changed_base<Test>
-        {
-            WIL_NOTIFYING_PROPERTY(int, MyProperty, 42);
-        };
-        auto test = winrt::make<Test>();
-        auto testImpl = winrt::get_self<Test>(test);
-        bool notified = false;
-        auto token = test.PropertyChanged([&](winrt::Windows::Foundation::IInspectable, winrt::Windows::UI::Xaml::Data::PropertyChangedEventArgs args)
+                struct Test : winrt::implements<Test, winrt::Windows::UI::Xaml::Data::INotifyPropertyChanged>, wil::notify_property_changed_base<Test>
+                {
+                    wil::single_threaded_notifying_property<int> MyProperty;
+                    Test() : INIT_NOTIFYING_PROPERTY(MyProperty, 42) {}
+                };
+                auto test = winrt::make<Test>();
+                auto testImpl = winrt::get_self<Test>(test);
+                bool notified = false;
+                auto token = test.PropertyChanged([&](winrt::Windows::Foundation::IInspectable, winrt::Windows::UI::Xaml::Data::PropertyChangedEventArgs args)
+                    {
+                        REQUIRE(args.PropertyName() == L"MyProperty");
+                        REQUIRE(testImpl->MyProperty() == 43);
+                        notified = true;
+                    });
+
+                testImpl->MyProperty(43);
+                REQUIRE(notified);
+                test.PropertyChanged(token);
+                REQUIRE(testImpl->MyProperty.Name() == L"MyProperty");
+            }
             {
-                REQUIRE(args.PropertyName() == L"MyProperty");
-                REQUIRE(testImpl->MyProperty() == 43);
-                notified = true;
-            });
-
-        testImpl->MyProperty(43);
-        REQUIRE(notified);
-        test.PropertyChanged(token);
-    }
-    manager.Close();
+                struct Test : winrt::implements<Test, winrt::Windows::UI::Xaml::Data::INotifyPropertyChanged>, wil::notify_property_changed_base<Test>
+                {
+                    WIL_NOTIFYING_PROPERTY(int, MyProperty, 42);
+                };
+                auto test = winrt::make<Test>();
+                auto testImpl = winrt::get_self<Test>(test);
+                bool notified = false;
+                auto token = test.PropertyChanged([&](winrt::Windows::Foundation::IInspectable, winrt::Windows::UI::Xaml::Data::PropertyChangedEventArgs args)
+                    {
+                        REQUIRE(args.PropertyName() == L"MyProperty");
+                        REQUIRE(testImpl->MyProperty() == 43);
+                        notified = true;
+                    });
+
+                testImpl->MyProperty(43);
+                REQUIRE(notified);
+                test.PropertyChanged(token);
+            }
+            manager.Close();
+        });
+    controller.ShutdownQueueAsync().get();

```

Comment:
>get [](http://example.com/codeflow?start=36&length=3)

Is this guaranteed to complete in a reasonable amount of time?  get is a potentially infinite wait.

Reply:
I'll fold it with the WFSO.